### ISSUE TITLE
cmd/govim: add syntax-based motion functions

### ISSUE THREAD

**Issue by myitcv** [myitcv/govim#394]
This has long been an idea but never captured in an issue; was reminded of the idea because of a question in `#vim` on Slack.

Vim offers language agnostic motions, for example:

```
5. Text object motions                                  object-motions

                                                        (
(                       [count] sentences backward.  exclusive motion.

                                                        )
)                       [count] sentences forward.  exclusive motion.

                                                        {
{                       [count] paragraphs backward.  exclusive motion.

                                                        }
}                       [count] paragraphs forward.  exclusive motion.

                                                        ]]
]]                      [count] sections forward or to the next '{' in the
                        first column.  When used after an operator, then also
                        stops below a '}' in the first column.  exclusive
                        Note that exclusive-linewise often applies.
...
```

However, what is potentially more useful when coding in a specific language is motions oriented around the language's syntax.

For example:

* Next/previous declaration
* Next/previous comment
* ...

The list could easily go on.

Adding functions like `GOVIMMotionNextDecl` etc should be relatively easy. Whilst we don't currently parse buffer contents on changes from Vim (we simply send the delta to `gopls`), it would be trivial to do so.

**Comment by myitcv** [comment]
Prior work in this area: https://github.com/fatih/motion

**Comment by myitcv** [comment]
Made a start on this in #397 and #398

**Comment by myitcv** [comment]
An initial cut of this was implemented in #398.
